aw_datastore/storages/sqlite.py

```diff
@@ -0,0 +1,208 @@
+        endtime REAL NOT NULL,
+        datastr TEXT NOT NULL,
+        FOREIGN KEY (bucket) REFERENCES buckets(id)
+    )
+"""
+
+INDEX_EVENTS_TABLE = """
+    CREATE INDEX IF NOT EXISTS event_index ON events(bucket, starttime, endtime);
+"""
+
+
+class SqliteStorage(AbstractStorage):
+    sid = "sqlite"
+
+    def __init__(self, testing):
+        self.testing = testing
+        data_dir = get_data_dir("aw-server")
+
+        ds_name = self.sid + ('-testing' if testing else '')
+        filename = ds_name + ".v{}".format(LATEST_VERSION) + '.db'
+        filepath = os.path.join(data_dir, filename)
+        new_db_file = not os.path.exists(filepath)
+        self.conn = sqlite3.connect(filepath)
+        logger.info("Using database file: {}".format(filepath))
+
+        # Create tables
+        self.conn.execute(CREATE_BUCKETS_TABLE)
+        self.conn.execute(CREATE_EVENTS_TABLE)
+        self.conn.execute(INDEX_EVENTS_TABLE)
+        self.commit()
+
+        if new_db_file:
+            logger.info("Created new SQlite db file")
+            from aw_datastore import check_for_migration
+            check_for_migration(self, ds_name, LATEST_VERSION)
+
+    def commit(self):
+        # Useful for debugging and trying to lower the amount of
+        # unnecessary commits
+        self.conn.commit()
+
+    def buckets(self):
+        buckets = {}
+        c = self.conn.cursor()
+        for row in c.execute("SELECT * FROM buckets"):
+            buckets[row[0]] = {
+                "id": row[0],
+                "name": row[1],
+                "type": row[2],
+                "client": row[3],
+                "hostname": row[4],
+                "created": row[5],
+            }
+        return buckets
+
+    def create_bucket(self, bucket_id: str, type_id: str, client: str,
+                      hostname: str, created: str, name: Optional[str] = None):
+        self.conn.execute("INSERT INTO buckets VALUES (?, ?, ?, ?, ?, ?)",
+            [bucket_id, name, type_id, client, hostname, created])
+        self.commit();
+        return self.get_metadata(bucket_id)
+
+    def delete_bucket(self, bucket_id: str):
+        self.conn.execute("DELETE FROM events WHERE bucket = ?", [bucket_id])
+        self.conn.execute("DELETE FROM buckets WHERE id = ?", [bucket_id])
+        self.commit()
+
+    def get_metadata(self, bucket_id: str):
+        c = self.conn.cursor()
+        res = c.execute("SELECT * FROM buckets WHERE id = ?", [bucket_id])
+        row = res.fetchone()
+        bucket = {
+            "id": row[0],
+            "name": row[1],
+            "type": row[2],
+            "client": row[3],
+            "hostname": row[4],
+            "created": row[5],
+        }
+        return bucket
+
+    def insert_one(self, bucket_id: str, event: Event) -> Event:
+        c = self.conn.cursor()
+        starttime = event.timestamp.timestamp() * 1000000
+        endtime = starttime + (event.duration.total_seconds() * 1000000)
```

Comment:
Would be nice if this could be simplified. 
``` python3
endtime = event.timestamp + event.duration
```
then like I describe below, register an adapter
``` python3
sqlite3.register_adapter(datetime.datetime, lambda ts: ts.timestamp()) // Multiply with 1000000 if you want or just save as REAL
```